subprojects/core/src/test/groovy/org/gradle/api/internal/changedetection/state/IndexedNormalizedFileSnapshotTest.groovy

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.changedetection.state
+
+import spock.lang.Specification
+
+class IndexedNormalizedFileSnapshotTest extends Specification {
+
+    def content = MissingFileContentSnapshot.instance
+
+    def "sorts by normalized path in lexicographical order"() {
+        given:
+        def unsorted = [
+            new IndexedNormalizedFileSnapshot("/baz/fizz", 4, content),
+            new IndexedNormalizedFileSnapshot("/foo/bar", 4, content),
+            new NonNormalizedFileSnapshot("/fizz", content),
+            new IndexedNormalizedFileSnapshot("/baz/bar", 4, content),
```

Comment:
❌ Can we add some more indexed paths starting at different positions and also on path having another path as a prefix (e.g. `prefix`, `prefixa` and `prefixb`)?